src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -4288,14 +4279,29 @@ int32_t NaClSysExecve(struct NaClAppThread *natp, char const *path, char *const
   }
 
   /* reset permissions to executable */
+  NaClLog(2, "Setting child permissions to executable\n");
   NaClVmmapChangeProt(&nap_child->mem_map, tramp_pnum, tramp_npages, PROT_RX);
+  NaClLog(2, "Setting parent permissions to executable\n");
   NaClVmmapChangeProt(&nap->mem_map, tramp_pnum, tramp_npages, PROT_RX);
+
+  NaClLog(2, "Child mprotect\n");
   if (NaClMprotect((void *)child_start_addr, tramp_size, PROT_RX) == -1) {
+    NaClLog(LOG_FATAL, "%s\n", "child vmmap page NaClMprotect failed!");
   }
+  NaClLog(2, "Parent mprotect\n");
+
   if (NaClMprotect((void *)parent_start_addr, tramp_size, PROT_RX) == -1) {
     NaClLog(LOG_FATAL, "%s\n", "parent vmmap page NaClMprotect failed!");
   }
 
+  /* Copy fd table in SafePOSIX */
+  NaClXMutexLock(&nap->mu); 
+  NaClXMutexLock(&nap_child->mu); 
+  NaClLog(2, "Copying fd table in SafePOSIX\n");
+  lind_fork(nap_child->cage_id, nap->cage_id);
+  NaClXMutexUnlock(&nap_child->mu);
+  NaClXMutexUnlock(&nap->mu);
+
```

Comment:
This is the meat of the issue, right?

Reply:
This is the meat of the exec fds in safeposix. The meat of the real issue was the weird cache_table that I removed.